bugFix #652 and feat(AdminFilter) hide menu when and admin start scro…

OlehFilyak opened the pull request:
Pull Request Template
 

### Type of Pull Request *
- [] CHANGE (fix or feature that would cause existing functionality to not work as expected)
- [ ] FEATURE (non-breaking change which adds functionality)
- [x] BUGFIX (non-breaking change which fixes an issue)
- [ ] ENHANCEMENT  (non-breaking change which improves existing functionality)
- [ ] NONE (if none of the other choices apply. For example, tooling, build system, CI, docs, etc.)

### Related links
 
**Issue link**
[bugFix [#652](https://github.com/ita-social-projects/dokazovi-requirements/issues/652)]
 
### Description *
Fix the bug by adding a new feature - when the admin starts scrolling down we hide the the menu

OlehFilyak committed:
bugFix #652 and feat(AdminFilter) hide menu when and admin start scroll down

Implement logic hiding menu when the admin starts scrolling down

skyboyer commented on src/views/Profile/AdminTable/AdminFilter.tsx:
```diff
@@ -51,6 +50,30 @@ export const AdminFilter: React.FC<IMaterialsFilter> = ({
   const classes = useStyles();
   const { t } = useTranslation();
+  const [open, setOpen] = useState(false);
+  const selectRef = useRef(null);
```
why do we need this ref? I don't see our code to consume it

skyboyer commented on src/views/Profile/AdminTable/AdminFilter.tsx:
```diff
@@ -51,6 +50,30 @@ export const AdminFilter: React.FC<IMaterialsFilter> = ({
+    };
+  }, [open]);
+
+  const handleOpen = () => {
```
if you wrap `handleOpen` and `handleClose` into `useCallback()` you could use `handleClose` in the `useEffect` above without risk to fall into infinite rerender

skyboyer commented on src/views/Profile/AdminTable/AdminFilter.tsx:
```diff
@@ -51,6 +50,30 @@ export const AdminFilter: React.FC<IMaterialsFilter> = ({
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if (open) {
```
but do we really need this check? I guess, calling `setOpen(false)` will work fine even if dropdown is already closed. No?

OlehFilyak committed:
bugfix_652

remove useRef
remove unnecessary checking
add useCallback to handleOpen and handleClose

skyboyer commented on src/views/Profile/AdminTable/AdminFilter.tsx:
```diff
@@ -51,29 +51,18 @@ export const AdminFilter: React.FC<IMaterialsFilter> = ({
-        setOpen(false);
-      }
-    };
+    const handleScroll = () => handleClose();
```
you could directly use `handleClose` but using `handleScroll` as a wrapper around it is also fine

in my opinion
